valle/data/tokenizer.py

```diff
@@ -30,6 +30,45 @@
 from phonemizer.separator import Separator
 
 
+class PypinyinBackend:
+    """PypinyinBackend for Chinese. Most codes is referenced from espnet.  
+       For compliance with tokenize_text, return need to be in [str].
+       In Tokenize Text process, phonemes also should be phonemes.split("_") so as to 
+       get the right symbols for 'unique_symbols.update(list(phonemes))'
+       
+       """
+    def __init__(
+        self,
+        g2p_type="pypinyin_g2p",
+    )-> None:
+        self.g2p_type=g2p_type
+    def phonemize(self,text,separator="", strip="", njobs=1):
+        # separator strip njobs are not used 
+        text=text[0].replace(" ","") # remove space
+        if self.g2p_type=="pypinyin_g2p":
+              from pypinyin import Style, pinyin
+              phones = [phone[0] for phone in pinyin(text, style=Style.TONE3)]
+              return [" ".join(phones)]
+        elif self.g2p_type=="pypinyin_g2p_phone":
+              from pypinyin import Style, pinyin
+              from pypinyin.style._utils import get_finals, get_initials
+
+              phones = [
+                  p
+                  for phone in pinyin(text, style=Style.TONE3)
+                  for p in [
+                      get_initials(phone[0], strict=True),
+                      get_finals(phone[0][:-1], strict=True) + phone[0][-1]
+                      if phone[0][-1].isdigit()
+                      else get_finals(phone[0], strict=True)
+                      if phone[0][-1].isalnum()
+                      else phone[0],
+                  ]
+                  # Remove the case of individual tones as a phoneme
+                  if len(p) != 0 and not p.isdigit()
+              ]
+              return [" ".join(phones)] #need to be return in [str] for compliance with tokenize_text
```

Comment:
we can refactor function `tokenize_text` to return list instead of str.
`TextTokenCollater` should be updated also.